### ISSUE TITLE
Tensorflow lite lstm/bilstm reset variables why needed ? Can we avoid reset variables. 

### ISSUE THREAD

**Issue by pranathibl** [tensorflow/tensorflow#47813]
Getting different outputs for multiple runs in tensorflow lite invoke when reset is not called. Why different output ? Is this step causes performance issue in multiple runs because of reset function extra step during every run. why interpreter.reset_all_variables() is needed.

`# -*- coding: utf-8 -*-
"""BiLSTMinput_Var_LastMode_layer.ipynb

Automatically generated by Colaboratory.

Original file is located at
    https://colab.research.google.com/drive/1xTKC1yQ4S016p_yFFFm91TCJlHj7nnDf
"""

import numpy as np
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers

model = keras.Sequential()

# Add a LSTM layer with 128 internal units.
model.add(layers.Bidirectional(layers.LSTM(input_shape=(None,20),units= 128, return_sequences = False)))

# Step 2: Train & Evaluate the model.

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy']);
x_test = np.zeros([3000,4,20], dtype=np.float32);
x_train = np.zeros([3000,4,20], dtype=np.float32);
y_test =  np.zeros([3000, 1,], dtype=np.float32);
y_train = np.zeros([3000, 1,], dtype=np.float32);

model.fit(x_train, y_train, epochs=5)
model.evaluate(x_test, y_test)

model.summary()

run_model = tf.function(lambda x: model(x))
# This is important, let's fix the input size.
BATCH_SIZE = 1
STEPS = 2
INPUT_SIZE = 20

concrete_func = run_model.get_concrete_function(
    tf.TensorSpec([BATCH_SIZE, STEPS, INPUT_SIZE], model.inputs[0].dtype))

# model directory.
MODEL_DIR = "keras_bi_lstm_new"
model.save(MODEL_DIR, save_format="tf", signatures=concrete_func)

converter = tf.lite.TFLiteConverter.from_saved_model(MODEL_DIR)
tflite_model = converter.convert()

with open('bilstmmodel_1_2_20.tflite', 'wb') as f:
  f.write(tflite_model)

# Run the model with TensorFlow Lite
interpreter = tf.lite.Interpreter(model_content=tflite_model)
#interpreter = tf.lite.Interpreter(model_path="bilstmmodel_last_1_2_20.tflite")
interpreter.allocate_tensors()
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()

input_shape = input_details[0]['shape']

output_shape = output_details[0]['shape']
print(input_shape) 
print(output_shape)

myinp = np.loadtxt('img.txt')

myinp = np.resize(myinp,input_shape);
myinp = (np.float32(myinp) - 0) / 1
print(myinp)

interpreter.allocate_tensors()
interpreter.set_tensor(input_details[0]["index"], myinp)
interpreter.invoke()
result = interpreter.get_tensor(output_details[0]["index"])
print(result)
interpreter.invoke()
result = interpreter.get_tensor(output_details[0]["index"])
print(result)

interpreter.reset_all_variables()
interpreter.invoke()
result = interpreter.get_tensor(output_details[0]["index"])
print(result)
print(result.shape)`

**Comment by Saduf2019** [comment]
@pranathibl 
We see that the issue template has not been filled, could you please do so as it helps us analyse the issue [tf version, steps followed before you ran into this error or stand alone code to reproduce the issue faced]

**Comment by abattery** [comment]
@renjie-liu could you take a look at this?

**Comment by renjie-liu** [comment]
Yes, the reset variable call is expected, the lstm kernel in tflite is stateful (meaning the states will be mutated and stayed there so you don't need to copy and pass in again).

**Comment by pranathibl** [comment]
Is there any way to avoid based on network type. For CNN's we dont need that right ?

**Comment by renjie-liu** [comment]
Yes, for CNN network you don't need to do that
